oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/registration/ClientRegistration.java

```diff
@@ -655,6 +655,10 @@ private ClientRegistration create() {
 			clientRegistration.clientName = StringUtils.hasText(this.clientName) ? this.clientName
 					: this.registrationId;
 			clientRegistration.clientSettings = this.clientSettings;
+			if (this.clientSettings.requireProofKey) {
```

Comment:
Please remove this since `clientRegistration.clientSettings` is set in previous line with `this.clientSettings` and only the `clientSettings` should be used.